pkg/codegen/programs.md

```diff
@@ -0,0 +1,16 @@
+(programgen)=
+# Programs
+
+Programs may be generated from a program written in a supported source language.
+This process, often referred to as "programgen", exists primarily to support the
+[`pulumi convert`](https://www.pulumi.com/docs/cli/commands/pulumi_convert/)
+command, though it is also used in other places such as [language conformance
+testing](language-conformance-tests). Programgen begins by using a [converter
+plugin](plugins) to convert from a source language into [Pulumi Configuration
+Language](pcl) (PCL), before generating a program in the target language from
+this PCL definition. The PCL-to-target-language conversion is exposed by a
+[language host](language-hosts)'s [](pulumirpc.LanguageRuntime.GenerateProgram)
```

Comment:
Probably worth explaining the difference between GenerateProgram and GenerateProject here.
Program only generates the code files (e.g. .ts/.cs) while Project will also generate the Pulumi.yaml file and any project meta files (e.g. package.json/.csproj)